StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift

```diff
@@ -8,8 +8,54 @@
 import Foundation
 @_spi(STP) import StripeCore
 
-protocol FinancialConnectionsAPIClient {
+class FinancialConnectionsAPIClient {
+    let backingAPIClient: STPAPIClient
 
+    var isLinkWithStripe: Bool = false
+    var consumerPublishableKey: String?
+    var consumerSession: ConsumerSessionData?
+
+    static let shared = FinancialConnectionsAPIClient(apiClient: .shared)
+
+    init(apiClient: STPAPIClient) {
+        self.backingAPIClient = apiClient
+    }
```

Comment:
⚠️ 

this part is likely the most critical part for this PR to think about

something feels a bit off about instantiation right now, and I wonder how we can clean this up so we can be "100%" sure that there is only one usage/instance of `FinancialConnectionsAPIClient`

---

it seems like `FinancialConnectionsAPIClient.shared` is only used in one place - SwiftUI previews? which maybe means we don't need it

---

then `FinancialConnectionsAPIClient(apiClient: apiClient)` is instantiated in `HostController`, but "the journey of using the API client" begins in `FinancialConnectionsSheet` - can we move the instantiation "up the stack to `FinancialConnectionsSheet` and then ensure it's 'never' done anywhere else? 

For example, maybe that also means that `STPAPIClient` should never be declared anywhere BUT `FinancialConnectionsSheet` (and inside of the new `FinancialConnectionsAPIClient` class).  (likely a grep should show that)

---

then another thing is that for some reason `apiClient` is public...what happens if a merchant swaps out `apiClient`? maybe it makes sense to have the instantiation of `FinancialConnectionsAPIClient` be done inside of `FinancialConnectionsSheet`, AND make sure that we re-instantiate it incase anyone changes `apiClient: STPAPIClient` with a public variable

this is very similar to what's happening with `APIVersion.configureFinancialConnectionsAPIVersion(apiClient: apiClient)`: its called in two places: `init` AND `didSet` of apiClient. likely the same has to be done for `FinancialConnectionsAPIClient`?

(small side-note: it does seem like we 'can't' "guarantee" that users of `FinancialConnectionsSheet` don't swap out the api client _after_ we already presented the sheet, but that's OK...that's already true today so we can just focus on the narrow/common use-case and treat that as user error if that ever does happen)

Reply:
Good points - I'll move the initialization of `FinancialConnectionsAPIClient` to `FinancialConnectionsSheet`!

Thinking about how it should be setup on the `FinancialConnectionsSheet` (stored property vs computed property vs initialized inline when we initialized the `HostController`) - I don't think any approach is better than any other, knowing it needs to be re-initialized whenever `apiClient` changes. So, going with the "simplest approach", I'll probably just initialize it inline where `HostController` is created!